components/kyma-environment-broker/internal/process/provisioning/inject_btp_operator_credentials_step.go

```diff
@@ -0,0 +1,143 @@
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	updateSecretBackoff = 10 * time.Second
+	secretName          = "sap-btp-manager"
+	secretNamespace     = "kyma-system"
+)
+
+var labels = map[string]string{"app.kubernetes.io/managed-by": "kcp-kyma-environment-broker"}
+var annotations = map[string]string{"Warning": "This secret is generated. Do not edit!"}
+
+type InjectBTPOperatorCredentialsStep struct {
+	operationManager  *process.OperationManager
+	k8sClientProvider func(kubeconfig string) (client.Client, error)
+}
+
+func NewInjectBTPOperatorCredentialsStep(os storage.Operations, k8sClientProvider func(kcfg string) (client.Client, error)) *InjectBTPOperatorCredentialsStep {
+	return &InjectBTPOperatorCredentialsStep{
+		operationManager:  process.NewOperationManager(os),
+		k8sClientProvider: k8sClientProvider,
+	}
+}
+
+func (s *InjectBTPOperatorCredentialsStep) Name() string {
+	return "Inject_BTP_Operator_Credentials"
+}
+
+func (s *InjectBTPOperatorCredentialsStep) Run(operation internal.Operation, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+
+	if operation.RuntimeID == "" {
+		log.Error("Runtime ID is empty")
+		return s.operationManager.OperationFailed(operation, "Runtime ID is empty", nil, log)
+	}
+
+	if operation.K8sClient == nil {
+		log.Error("kubernetes client not set")
+		return s.operationManager.OperationFailed(operation, "kubernetes client not set", nil, log)
+	}
+
+	clusterID := operation.InstanceDetails.ServiceManagerClusterID
+	if clusterID == "" {
+		clusterID = uuid.NewString()
+		updatedOperation, backoff, err := s.operationManager.UpdateOperation(operation, func(op *internal.Operation) {
+			op.InstanceDetails.ServiceManagerClusterID = clusterID
+		}, log)
+		if err != nil {
+			log.Errorf("failed to update operation: %w", err)
+		}
+		if backoff != 0 {
+			log.Error("cannot save cluster ID")
+			return updatedOperation, backoff, nil
+		}
+	}
+
+	secret := s.prepareSecret(operation.ProvisioningParameters.ErsContext.SMOperatorCredentials, clusterID)
+
+	if err := s.createOrUpdateSecret(operation.K8sClient, secret, log); err != nil {
+		err = kebError.AsTemporaryError(err, "failed create/update of the secret")
+		return operation, updateSecretBackoff, nil
+	}
+	return operation, 0, nil
+}
+
+func (s *InjectBTPOperatorCredentialsStep) prepareSecret(credentials *internal.ServiceManagerOperatorCredentials, clusterID string) *apicorev1.Secret {
+	return &apicorev1.Secret{
+		TypeMeta: v1.TypeMeta{Kind: "Secret"},
+		ObjectMeta: v1.ObjectMeta{
+			Name:        secretName,
+			Namespace:   secretNamespace,
+			Labels:      labels,
+			Annotations: annotations,
+		},
+		StringData: map[string]string{
+			"clientid":       credentials.ClientID,
+			"clientsecret":   credentials.ClientSecret,
+			"sm_url":         credentials.ServiceManagerURL,
+			"tokenurl":       credentials.URL,
+			"tokenurlsuffix": credentials.XSAppName,
```

Comment:
`tokenurlsuffix` is not required, it stays the same as in upstream BTP Service Operator chart. Can be removed from the Secret.

Reply:
https://github.com/SAP/sap-btp-service-operator#namespace-secret-structure

Reply:
Changed and the change pushed.